controllers/velero.go

```diff
@@ -664,174 +673,87 @@ func getAppLabels(instanceName string) map[string]string {
 	return labels
 }
 
-// Get Velero Resource Requirements
-func (r *DPAReconciler) getVeleroResourceReqs(dpa *oadpv1alpha1.DataProtectionApplication) (corev1.ResourceRequirements, error) {
-
-	// Set default values
-	ResourcesReqs := corev1.ResourceRequirements{
-		Requests: corev1.ResourceList{
-			corev1.ResourceCPU:    resource.MustParse("500m"),
-			corev1.ResourceMemory: resource.MustParse("128Mi"),
-		},
-	}
-
-	if dpa != nil && dpa.Spec.Configuration != nil && dpa.Spec.Configuration.Velero != nil && dpa.Spec.Configuration.Velero.PodConfig != nil {
-		// Set custom limits and requests values if defined on VELERO Spec
-		if dpa.Spec.Configuration.Velero.PodConfig.ResourceAllocations.Requests.Cpu() != nil && dpa.Spec.Configuration.Velero.PodConfig.ResourceAllocations.Requests.Cpu().Value() != 0 {
-			parsedQuantity, err := resource.ParseQuantity(dpa.Spec.Configuration.Velero.PodConfig.ResourceAllocations.Requests.Cpu().String())
-			ResourcesReqs.Requests[corev1.ResourceCPU] = parsedQuantity
-			if err != nil {
-				return ResourcesReqs, err
-			}
+// getResourceListFrom get the values of cpu, memory, storage and ephemeral-storage from
+// input into fallback.
+func getResourceListFrom(input corev1.ResourceList, fallback corev1.ResourceList) (*corev1.ResourceList, error) {
+	if input.Cpu() != nil && input.Cpu().Value() != 0 {
+		parsedQuantity, err := resource.ParseQuantity(input.Cpu().String())
+		if err != nil {
+			return nil, err
 		}
-
-		if dpa.Spec.Configuration.Velero.PodConfig.ResourceAllocations.Requests.Memory() != nil && dpa.Spec.Configuration.Velero.PodConfig.ResourceAllocations.Requests.Memory().Value() != 0 {
-			parsedQuantity, err := resource.ParseQuantity(dpa.Spec.Configuration.Velero.PodConfig.ResourceAllocations.Requests.Memory().String())
-			ResourcesReqs.Requests[corev1.ResourceMemory] = parsedQuantity
-			if err != nil {
-				return ResourcesReqs, err
-			}
+		fallback[corev1.ResourceCPU] = parsedQuantity
+	}
+	if input.Memory() != nil && input.Memory().Value() != 0 {
+		parsedQuantity, err := resource.ParseQuantity(input.Memory().String())
+		if err != nil {
+			return nil, err
 		}
-
-		if dpa.Spec.Configuration.Velero.PodConfig.ResourceAllocations.Limits.Cpu() != nil && dpa.Spec.Configuration.Velero.PodConfig.ResourceAllocations.Limits.Cpu().Value() != 0 {
-			if ResourcesReqs.Limits == nil {
-				ResourcesReqs.Limits = corev1.ResourceList{}
-			}
-			parsedQuantity, err := resource.ParseQuantity(dpa.Spec.Configuration.Velero.PodConfig.ResourceAllocations.Limits.Cpu().String())
-			ResourcesReqs.Limits[corev1.ResourceCPU] = parsedQuantity
-			if err != nil {
-				return ResourcesReqs, err
-			}
+		fallback[corev1.ResourceMemory] = parsedQuantity
+	}
+	if input.Storage() != nil && input.Storage().Value() != 0 {
```

Comment:
Remove?

![Screenshot from 2024-08-02 09-59-59](https://github.com/user-attachments/assets/430e69e1-fa2c-40a9-b6c2-6d9508616ace)

![Screenshot from 2024-08-02 10-00-45](https://github.com/user-attachments/assets/2d3a849e-5f83-4243-89f8-e25b063a3191)

Reply:
References
`https://github.com/pachyderm/pachyderm/issues/4038`
https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#setting-requests-and-limits-for-local-ephemeral-storage